libcob/cconv.c

```diff
@@ -471,3 +472,111 @@ cob_get_collation_by_name (const char *col_name,
 
 	return col_id;
 }
+
+static int
+cob_convert_hex_digit (char h)
+{
+	if (h >= '0' && h <= '9') return h - '0';
+	else if (h >= 'A' && h <= 'F') return 10 + h - 'A';
+	else if (h >= 'a' && h <= 'f') return 10 + h - 'a';
+	else return -1;
+}
+
+static int
+cob_convert_hex_byte (const char *h)
+{
+	int d1 = cob_convert_hex_digit (h[0]);
+	int d2 = cob_convert_hex_digit (h[1]);
+	if (d1 < 0 || d2 < 0) {
+		return -1;
+	} else {
+		return d1 * 16 + d2;
+	}
+}
+
+/* Note: cobc uses this function to create tables that are embedded in the
+   generated code, and we also make it available in libcob so that external
+   tools that make use of the library can use this feature */
+int
+cob_load_collation (const char *col_name,
+		    cob_u8_t *ebcdic_to_ascii,
+		    cob_u8_t *ascii_to_ebcdic)
+{
+	char hex[512];
+	int n, i;
+	FILE *f;
+        char *config_dir;
+	char filename[COB_FILE_BUFF];
+	const char *last_col_name = NULL;
+
+	if (col_name[0] == '.' || col_name[0] == SLASH_CHAR
+#ifdef _WIN32
+ || col_name[0] != '\0' && col_name[1] == ':'
+#endif
+	   ) {
+		/* If it's a path, use it as-is */
+		snprintf (filename, (size_t)COB_FILE_MAX, "%s", col_name);
+	} else {
+		/* Otherwise, prepend the config dir and append the .ttbl suffix */
+		config_dir = getenv ("COB_CONFIG_DIR");
+		snprintf (filename, (size_t)COB_FILE_MAX, "%s%c%s.ttbl",
+			  config_dir ? config_dir : COB_CONFIG_DIR, SLASH_CHAR, col_name);
+	}
+
+	f = fopen(filename, "rb");
```

Comment:
Is there a reason to read the file in binary mode? Otherwise please drop the "b".